keps/sig-scheduling/20180409-scheduling-framework.md

```diff
@@ -198,49 +197,53 @@ post-filter extension point.
 
 ### Scoring
 
-These plugins are used to rank nodes that have passed the filtering phase. The
-scheduler will call each scoring plugin for each node. There will be a well
-defined range of integers representing the minimum and maximum scores. After the
-[normalize scoring](#normalize-scoring) phase, the scheduler will combine node
-scores from all plugins according to the configured plugin weights.
+These plugins have two phases:
 
-### Normalize scoring
-
-These plugins are used to modify scores before the scheduler computes a final
-ranking of Nodes. A plugin that registers for this extension point will be
-called with the [scoring](#scoring) results from the same plugin. This is called
-once per plugin per scheduling cycle.
+1. The first phase is called "score" which is used to rank nodes that have passed the
+filtering phase. The scheduler will call `Score` of each scoring plugin for each node. 
+There will be a well defined range of integers representing the minimum and maximum scores.
+2. The second phase is "normalize scoring" which is used to modify scores before the 
+scheduler computes a final ranking of Nodes. `NormalizeScore` is called once per plugin
+per scheduling cycle right after "score" phase. After the "normalize scoring" phase, the
+scheduler will combine node scores from all plugins according to the configured plugin 
+weights.
 
 For example, suppose a plugin `BlinkingLightScorer` ranks Nodes based on how
 many blinking lights they have.
 
 ```go
-func ScoreNode(_ *v1.Pod, n *v1.Node) (int, error) {
-   return getBlinkingLightCount(n)
+func (*BlinkingLightScorer) Score(pc *PluginContext, _ *v1.Pod, nodeName string) (int, *Status) {
+   return getBlinkingLightCount(nodeName)
 }
 ```
 
 However, the maximum count of blinking lights may be small compared to
-`NodeScoreMax`. To fix this, `BlinkingLightScorer` should also register for this
-extension point.
+`NodeScoreMax`. To fix this, `BlinkingLightScorer` should also implement `NormalizeScore`.
 
 ```go
-func NormalizeScores(scores map[string]int) {
+func (*BlinkingLightScorer) NormalizeScore(pc *PluginContext, scores NodeScoreList) *Status {
```

Comment:
NormalizeScore should receive a pod pointer as well. Now that [the PR](https://github.com/kubernetes/kubernetes/pull/80383/) for adding the extension is out, please use the same interface here in the doc.

Reply:
Done